headers/validation.h

```diff
@@ -166,11 +213,25 @@ struct UniformGenerator {
 	template <typename T>
 	T operator()(T low, T high, std::mt19937_64& rng) const {
 		static_assert(is_number_v<T>);
-		if constexpr(std::is_same_v<T, long long>)
-			return std::uniform_int_distribution<T>(low, high)(rng);
-		else
-			return std::uniform_real_distribution<T>(low, high)(rng);
+		if constexpr(std::is_same_v<T, long long>) {
+			assert(low <= high);
+			// since C++20 we can assume Two's Complement but any sane syste, used it before anyway
+			// rejection sampling is not as fast as possible
+			// but definetly unbiased
+			unsigned long long ul = static_cast<unsigned long long>(low);
+			unsigned long long uh = static_cast<unsigned long long>(high);
+			int shitfs = C20::countl_zero(uh - ul + 1ull) % std::numeric_limits<unsigned long long>::digits;
+			unsigned long long res;
+			do {
+				res = Random::bits64(rng) >> shitfs;
+			} while (res > uh - ul);
+			return static_cast<long long>(res + ul);
+		}	else {
```

Comment:
Can you make this `else if costexpr` for `long double`, and `[static_]assert(false)` in the remaining case

Reply:
sadly the solution [is ugly](https://stackoverflow.com/questions/38304847/constexpr-if-and-static-assert), so let's not